src/en/charms-upgrading.md

```diff
@@ -0,0 +1,28 @@
+Title: Upgrading applications
+
+# Upgrading applications
+
+An application is upgraded by upgrading its corresponding charm. Since a charm
+is always associated with a *channel* and a charm's version can fluctuate
+within that channel the following rules apply:
+
+- If the channel has a version **newer** than the deployed version then the latter
+  is **upgraded** to the former.
+- If the channel has a version **older** than the deployed version then the latter
+  is **downgraded** to the former.
+
+Channels can be specified with the `upgrade-charm` command. For example:
+
+```bash
+juju upgrade-charm mysql --channel edge
```

Comment:
We also need to mention the use of '--switch', '--path' and '--revision'

Reply:
The `switch` option does not upgrade an application. It *replaces* it with an entirely new one, to my amazement. Logically, this should be a part of charms-deploying since it is deploying a new application. Anyway, the linked developer-upgrade-charm.md file talks about it.

Reply:
As for `--path`, how does one discover the "original path" if the application was originally deployed from the Store?

> Note that the directory containing the charm must match what was originally used to deploy the charm as a superficial check that the updated charm is compatible.

Reply:
Nevertheless, these are all options to this command! If you don't want to do this now, we need to at least create an issue for it. These things need to be added to this page (even if they make no sense or are discussed elsewhere)

Reply:
Lots of pages do not cover every option every time a command is mentioned. That's what the commands.md page is for. I'll put it there anyways since you must see something I do not.

Do you know the answer to my `--path` question?